### PR TITLE
League stats abb

### PR DESCRIPTION
**Pull request by angelbyun**
Completed all testing and methods for League_Stat class; all methods and tests run and pass successfully. Will need to refactor all methods with helper methods.

### CODE REVIEW

File: lib/league_stats.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    total_goals_by_team.merge!(total_games_by_team) do |team_id, goals, games|
+      goals.to_f / games
+    end
+    @teams.each do |team|
+      if team.team_id == total_goals_by_team.key(total_goals_by_team.values.max)
+        return team.team_name
+      end
+    end
+  end
+
```
Comment (andrew-bingham1) [inline lib/league_stats.rb:39]: Good opportunities for refactoring above and below

File: lib/league_stats.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
+require 'csv'
+require 'team'
+require 'game_teams'
+require 'game'
```
Comment (thawley2) [inline lib/league_stats.rb:4]: I don't think this is required, but we can change that when we refactor for inheritance.

**Review by andrew-bingham1 — APPROVED** [review]
Nice job! I think some helper methods will be perfect for these methods

File: lib/league_stats.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+class LeagueStats
+  attr_reader :teams,
+              :game_teams,
+              :games
```
Comment (thawley2) [inline lib/league_stats.rb:9]: I don't think these are needed either since all the calls to these attributes are being done within this class.

**Review by thawley2 — APPROVED** [review]

File: lib/league_stats.rb (hunk 3 above)
Comment (CrowRising) [reply lib/league_stats.rb:9]: I  agree with you this is a habit and not necessarily a thoughtful piece of code

File: lib/league_stats.rb (hunk 2 above)
Comment (CrowRising) [reply lib/league_stats.rb:4]: Thank you for pointing this out

File: lib/league_stats.rb (hunk 1 above)
Comment (CrowRising) [reply lib/league_stats.rb:39]: Agreed! Looking forward to that!

**Pull request merged**